shared/packages/worker/src/worker/lib/expectationHandler.ts

```diff
@@ -57,7 +57,9 @@ export interface ExpectationHandler {
 	workOnExpectation: (
 		exp: Expectation.Any,
 		genericWorker: GenericWorker,
-		specificWorker: any
+		specificWorker: any,
+		/** An FYI, the work will be considered timed out if there are no progression reports within this interval*/
+		progressTimeout: number
```

Comment:
Is it reasonable to assume that all expectation will be able to provide periodical progress reports? I suspect that this will be the case for many of them, but if PM will be interacting with a remote system and having it do the operations, I expect it is quite likely that the API available may not provide progress reports or the progress report frequency may not be controllable. As such, wouldn't it make sense to move knowledge on the progressTimeout deeper into the expectations, so that they can make the judgement on what is an unacceptable amount of time to wait for a progress report?